src/sync/wait_group.rs

```diff
@@ -0,0 +1,131 @@
+use std::fmt;
+use std::process;
+use std::sync::Arc;
+
+use parking_lot::{Condvar, Mutex};
+
+/// Enables multiple threads to synchronize the beginning or end of some computation.
```

Comment:
Worth mentioning that, unlike `std::sync::Barrier`, `WaitGroup` allows threads that just drop it to finish executing before others finish waiting.

Reply:
Done.